internal/services/networkfunction/network_function_collector_policy_resource.go

```diff
@@ -0,0 +1,495 @@
+}
+
+type IngestionSourcesPropertiesFormatModel struct {
+	ResourceId string                       `tfschema:"resource_id"`
+	SourceType collectorpolicies.SourceType `tfschema:"source_type"`
+}
+
+type NetworkFunctionCollectorPolicyResource struct{}
+
+var _ sdk.ResourceWithUpdate = NetworkFunctionCollectorPolicyResource{}
+
+func (r NetworkFunctionCollectorPolicyResource) ResourceType() string {
+	return "azurerm_network_function_collector_policy"
+}
+
+func (r NetworkFunctionCollectorPolicyResource) ModelObject() interface{} {
+	return &NetworkFunctionCollectorPolicyModel{}
+}
+
+func (r NetworkFunctionCollectorPolicyResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return collectorpolicies.ValidateCollectorPolicyID
+}
+
+func (r NetworkFunctionCollectorPolicyResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile("^[a-zA-Z0-9]([-._a-zA-Z0-9]{0,78}[a-zA-Z0-9_])?$"),
+				"The name can contain only letters, numbers, periods (.), hyphens (-),and underscores (_), up to 80 characters, and it must begin with a letter or number and end with a letter, number or underscore.",
+			),
+		},
+
+		"location": commonschema.Location(),
+
+		"traffic_collector_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: azuretrafficcollectors.ValidateAzureTrafficCollectorID,
+		},
+
+		"emission_policy": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"emission_destination": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						MinItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"destination_type": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										string(collectorpolicies.DestinationTypeAzureMonitor),
+									}, false),
+								},
+							},
+						},
+					},
+
+					"emission_type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ForceNew: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(collectorpolicies.EmissionTypeIPFIX),
+						}, false),
+					},
+				},
+			},
+		},
+
+		"ingestion_policy": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"ingestion_source": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						MinItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"resource_id": {
```

Comment:
Do we know what resource IDs are supported here?

Reply:
Currently, it supports ExpressRoute circuits, but the service team says more types of resource IDs will be added soon. They suggest no limit the resource type here because the value does not rely on the change of swagger.